### ISSUE TITLE
Mortar rounds cause endless cookoff-like explosions - how to debug?

### ISSUE THREAD

**Issue by Magirot** [acemod/ACE3#4590]
Apologies if this isn't the right place for so ambiguous issues.

**Arma 3 Version:** `1.64` (stable)
**CBA Version:** `3.1.1` (stable)
**ACE3 Version:** `3.8.0` (stable)

**Mods:**
- `@CBA_A3`
- `@ace`
- `@ACRE2`
- `@CUP_Terrains_Core` and `@CUP_Terrains_Maps` 
- `@RHSAFRF`, `@RHSGREF` and `@RHSUSAF`
- `@ShackTac User Interface` 

**Description:**
Mortar rounds seem to cause endless explosions that grind the server to a halt (and before that make the game almost unplayable for anyone withing hearing distance). We're unsure if it can happen with a round on its own, or if armour is required as a target.

Some videos by one of our players (sorry for the silly descriptions):
https://youtu.be/Wd0b-8qT-Oo?t=5s
https://youtu.be/oPAKslfws1A?t=6s

- https://youtu.be/LLhpSvqylHM?t=45s
Here it looks like the mortar round alone causes the endless chain. Then again, it hasn't happened when we've just fired test rounds into the fields, and in this case armour was within short distance of where the mortar hit. No idea if that could be relevant.

Though RHS content was involved in each of the missions, the mortars themselves were vanilla content.

**Steps to reproduce:**
Not sure. I've tried to reproduce the issue in singleplayer with both our modset and with only `@CBA_A3` and `@ace` enabled, but haven't been successful.

We'll be testing it a bit next Thursday, but it would be interesting to know if there are any tips for proper debugging. Or if anyone else has had it happen.

**Where did the issue occur?**
- Dedicated

**Placed Modules:**
- Mk6 mortar module in one of the missions, none for the other two.

**RPT log file:**
- Server log would probably be more useful, but it'll be some time before I can access that. In the meantime here is a player .rpt from the latest case (lines 1729-2758 are from the relevant mission)

http://pastebin.com/vMbVpSsW

**Comment by commy2** [comment]
```
learn2code nubs, git gud
or it might be rhs
idk, idgaf
```

Who wrote that? He should get his ass in here and fix it. This project is open source after all.

**Comment by Magirot** [comment]
I think he was trying to parody something, but comes off as really idiotic, I agree.

**Comment by SilentSpike** [comment]
Possible locality issue somewhere (something making explosions on all machines rather than just one) is the only theory I currently have to explain this (assuming this is even caused by us). Would explain your failed single player reproduction testing.

However the fact that it seems to be caused by mortars makes me think it's unrelated to cookoff.

**Comment by commy2** [comment]
Needs to figure out which weapon was used a here.

**Comment by Glowbal** [comment]
> However the fact that it seems to be caused by mortars makes me think it's unrelated to cookoff.

Agreed. For the ammo cook off to occur, the mortar round would need to inherit from the CfgVehicles >> "AllVehicles" class and return ammunition (a lot of it it seems) for the script command `magazinesAmmo`.
See: https://github.com/acemod/ACE3/blob/master/addons/cookoff/XEH_postInit.sqf#L70

**Comment by Magirot** [comment]
The mortars were vanilla Csat and Independent mortars firing regular HE rounds. Thanks for the information so far, we'll focus on testing the mortars on Thursday then.

**Comment by KingZogg** [comment]
I can conform that this is an issue.
For my server mods used were.
ACE3 v3.8.8
CBA_A3 v3.1.1.161012
And Task force radio. v0.9.12 (10-10-2016)

My issue is just the same as the original OP.

Further to that.
One hit with an NLAW caused our server to stop.
Just sat there with everyone nibbling biscuits and scratching nuts.

Not sure what is going on .. but totally unplayable as it is.

**Comment by bux578** [comment]
@KingZogg NLAW: duplicate of #4571

**Comment by alganthe** [comment]
Could it be related to RHS fuckery with handleDamage ?

**Comment by SilentSpike** [comment]
If it's caused by the vehicles it could be, if it's being caused solely by the mortar rounds then it shouldn't be.

**Comment by KingZogg** [comment]
Confirmed.

NLAW issue is separate to the constant explosion issue.
Have seen both on the server with the versions mentioned above.

Only seen twice. No idea how to reproduce either issue.

Can someone please recommend a previous commit that works ?
Server is unplayable with the current version.

**Comment by alganthe** [comment]
Your safest bet is to remove cooking for now.

**Comment by KingZogg** [comment]
Will that require new bikeys and adjustments to the client side files also ?

**Comment by alganthe** [comment]
Just removing the pbo and bisign on both client and server should work fine.

**Comment by KingZogg** [comment]
So. Remove the .pbo and resign ?

**Comment by commy2** [comment]
>So. Remove the .pbo and resign ?

haha 🏳️

**Comment by Magirot** [comment]
I managed to replicate the issue in singleplayer with only ``@CBA_3`` and ``@ace`` (3.8.1). The difference was that I had forgot to set all the same settings that our server had.

**It seems it's caused by the explosive reflections system**. The chain explosion can trigger while firing anywhere, but the chances seem to increase when firing into areas with more objects.

Also it seems to take the game down even quicker in singleplayer, mine invariably crashed with a memory error in a dozen seconds or so.

**Steps to reproduce:**
1. Create a new mission, place down a mortar or two and some vehicles as targets
2. Enable the explosive reflections system ([description.ext](http://magirot.kapsi.fi/m/I/description.ext) for the mission)
3. Preview and fire around
4. Observe endless explosions and the game crashing

[The simple mission I used](http://magirot.kapsi.fi/m/I/mortartesting.Altis.zip)
[RPT](http://pastebin.com/FfwZWGAe) (not much to see here)

Just to confirm @KingZogg, did you have explosive reflections enabled as well?
